libraries/chain/include/eosio/chain/controller.hpp

```diff
@@ -19,6 +21,13 @@ namespace boost { namespace asio {
 
 namespace eosio { namespace vm { class wasm_allocator; }}
 
+namespace eosio::hotstuff {
+   struct hs_message;
+   struct finalizer_state;
+   enum class hs_message_warning;
+   using bls_pub_priv_key_map_t = std::map<std::string, std::string>;
+}
```

Comment:
I don't think there would be any ambiguity if these were defined in `eosio::chain`